### PR TITLE
release-4.0.0 (HDS-2124) New tag component

### PR DESCRIPTION
**Pull request by mrTuomoK**
## Description

Renewed the Tag component according to the new designs.

https://city-of-helsinki.github.io/hds-demo/preview_1237/
https://city-of-helsinki.github.io/hds-demo/preview_1237/storybook/react
https://city-of-helsinki.github.io/hds-demo/preview_1237/storybook/core

## Related Issue

[HDS-2124](https://helsinkisolutionoffice.atlassian.net/browse/HDS-2124)
[HDS-2217](https://helsinkisolutionoffice.atlassian.net/browse/HDS-2217)
[HDS-2126](https://helsinkisolutionoffice.atlassian.net/browse/HDS-2126)

## How Has This Been Tested?

- local machine running all tests

## Add to changelog
- [x] Added needed line to changelog

[HDS-2124]: https://helsinkisolutionoffice.atlassian.net/browse/HDS-2124?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[HDS-2217]: https://helsinkisolutionoffice.atlassian.net/browse/HDS-2217?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[HDS-2126]: https://helsinkisolutionoffice.atlassian.net/browse/HDS-2126?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/react/src/components/tag/Tag.tsx [hunk 1]
```diff
@@ -6,138 +6,177 @@ import { IconCross } from '../../icons';
...
-   * Callback function fired when the delete icon is clicked. If set, a delete button will be shown.
+   * Callback function fired when the tag is clicked. If set, a delete button will be shown.
    */
-  onDelete?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
-  /**
-   * Sets the role of the tag when it's clickable. Uses 'link' by default.
-   * @deprecated Usage will change in the next major release.
-   */
-  role?: 'link' | 'button';
+  onDelete?: (event: React.MouseEvent<HTMLDivElement, MouseEvent> | React.KeyboardEvent<HTMLDivElement>) => void;
```
Comment (NikoHelle) [inline packages/react/src/components/tag/Tag.tsx:66]: The old text is kind of misleading: "a delete button will be shown". Should be "delete icon"?

Oh, it is "delete icon" in docs.

**Review by NikoHelle — CHANGES_REQUESTED** [review]
Looks good! One typo and one text improvement found.

File: packages/react/src/components/tag/Tag.tsx [hunk 2]
```diff
@@ -6,138 +6,177 @@ import { IconCross } from '../../icons';
...
    */
-  labelProps?: React.ComponentPropsWithoutRef<'span'>;
+  multiline?: boolean;
   /**
    * Callback function fired when the tag is clicked. If set, the tag will be clickable.
    */
   onClick?: (event: React.MouseEvent<HTMLDivElement, MouseEvent> | React.KeyboardEvent<HTMLDivElement>) => void;
   /**
-   * Callback function fired when the delete icon is clicked. If set, a delete button will be shown.
+   * Callback function fired when the tag is clicked. If set, a delete button will be shown.
```
Comment (NikoHelle) [inline packages/react/src/components/tag/Tag.tsx:64]: The old text is kind of misleading: "a delete button will be shown". Should be "delete icon"?

Oh, it is "delete icon" in docs.

Comment (mrTuomoK) [reply packages/react/src/components/tag/Tag.tsx:64]: Rephrased it a bit on the documentation site

File: packages/core/src/components/tag/tag.stories.js [hunk 3]
```diff
@@ -4,68 +4,167 @@ export default {
...
   </div>
 `;
 
-export const LargeTagDeletable = () => `
-  <div class="hds-tag hds-tag--large">
-    <span class="hds-tag__label" aria-hidden="true">Label</span>
-    <button aria-label="Delete item: Label" class="hds-tag__delete-button button-reset">
-        <span aria-hidden="true" class="hds-icon hds-icon--cross"></span>
-    </button>
+const LontTextTags = (sizeClass) => `
```
[the remark this answers is not in the document]
Comment (mrTuomoK) [reply packages/core/src/components/tag/tag.stories.js]: fixed

**Review by NikoHelle — APPROVED** [review]

**Pull request merged**